src/Formatter/CurlCommandFormatter.php

```diff
@@ -36,7 +36,10 @@ public function formatRequest(RequestInterface $request)
 
         $body = $request->getBody();
         if ($body->getSize() > 0) {
-            if ($body->isSeekable()) {
+            // escapeshellarg argument max length on Windows, but impractical to use either way
+            if ($body->getSize() > 8192) {
+                $data = '[too long stream omitted]';
+            } elseif ($body->isSeekable()) {
                 $data = $body->__toString();
```

Comment:
could we have a body that does not know its size? then we might need to check again how long $data is?

Reply:
on line 38 your have 
```php
if ($body->getSize() > 0) {
```

So I guess passing that point, it knows it size. But I can add another check just in case

Reply:
oh, i missed that. no then its fine!